app/views/file_upload.py

```diff
@@ -0,0 +1,52 @@
+from django.shortcuts import render, redirect
+
+from app.models import ColumnMetadata, TableMetadata
+
+
+def file_upload(request, table_id):
+    if request.method == "GET":
+        return render_file_upload_page(request, table_id, "")
+
+    upload_error = extract_columns_and_get_error(table_id, request.FILES)
+    if upload_error:
+        return render_file_upload_page(request, table_id, upload_error)
+
+    return redirect("/edit-table-columns/" + str(table_id))
+
+
+def render_file_upload_page(request, table_id, upload_error):
+    table_metadata = TableMetadata.objects.get(pk=table_id)
+    file_upload_data = {"table_name": table_metadata.name, "upload_error": upload_error}
+    return render(request, "file-upload.html", file_upload_data)
+
+
+def extract_columns_and_get_error(table_id, files):
+    # File is missing
+    uploaded_file = files.get("uploaded_file", None)
+    if not uploaded_file:
+        return "The file is missing!"
+
+    if not uploaded_file.name.endswith(".csv"):
+        return "Unsupported file format: ." + uploaded_file.name.split(".")[-1]
```

Comment:
```suggestion
    if not uploaded_file.name.endswith(".csv"):
        raise StopUpload(True)
```

Same as above. Unless there is a really strong reason to use `return` instead.

Reply:
I need to return a error message to the user, but doing this by rasing an error is a great idea